src/data/seq/seq.lean

```diff
@@ -643,6 +652,12 @@ begin
     simp only [seq.nth_tail _ n, (hyp $ n + 1)] }
 end
 
+lemma cons_injective (x : α) : function.injective (cons x) :=
```

Comment:
Hmm... these lemmas could be moved much earlier in the file if #15830 were merged first. So I think it's best if that's made the dependency instead of the other way around.